code/obj/item/clothing/suits.dm

```diff
@@ -28,96 +28,128 @@
 		setProperty("heatprot", 5)
 		setProperty("meleeprot", 2)
 
-/obj/item/clothing/suit/hoodie
+/obj/item/clothing/suit/hooded
+	name = "You shouldn't see this!"
+	desc = "Call 180-IMCODER"
+	uses_multiple_icon_states = TRUE
+	var/hood_style = null
+	var/hooded = FALSE
+	abilities = list(/obj/ability_button/hood_toggle)
+
+	attack_self(mob/user as mob)
+		..()
+		if (hooded)
+			src.unhood()
+		else
+			src.hood()
+
+	proc/hood()
+		if (src.hood_style)
+			src.icon_state = "[src.hood_style]-up"
+			usr.set_clothing_icon_dirty()
+			src.over_hair = TRUE
+			src.body_parts_covered = HEAD|TORSO|ARMS
+		usr.visible_message("[usr] flips [his_or_her(usr)] [src.name]'s hood up.",\
+		"You flip your [src.name]'s hood up.")
+		src.hooded = TRUE
+
+	proc/unhood()
+		if (src.hood_style)
+			src.icon_state = src.hood_style
+			usr.set_clothing_icon_dirty()
+			src.over_hair = FALSE
+			src.body_parts_covered = TORSO|ARMS
+		usr.visible_message("[usr] flips [his_or_her(usr)] [src.name]'s hood down.",\
+		"You flip your [src.name]'s hood down'.")
+		src.hooded = FALSE
```

Comment:
I would maybe advise making a hood-toggle proc instead that only toggles the hood, no message or anything, just something like src.hooded = !src.hooded and a call to UpdateIcon()?

That brings me to my next topic, maybe this whole thing should override update_icon()?

Then both the button and the attack_self proc can just call the hoodie toggle and give the person doing the toggling the message, that way you also avoid having to use usr.

You can then also just call UpdateIcon() in the New() of hoodie if necessary, to set the colored icon_state correctly. This way you avoid code duplication regarding the different icon_state settings.

Reply:
It has been done, thanks for the help!